library/ajax/upload.php

```diff
@@ -51,18 +63,62 @@
 $patient_id = filter_input(INPUT_GET, 'patient_id');
 $category_id = filter_input(INPUT_GET, 'parent_id');
 
+if ($isPortal ?? false) {
+    $owner = $GLOBALS['userauthorized'];
+    $files = getMultiple();
+    if (count($files["file"] ?? []) > 0) {
+        $messageService = new MessageService();
+        $data = [];
+        $note['groupname'] = 'Default';
+        // will send to all auth'ed portal users
+        $note['to'] = 'portal-user';
+        $note['from'] = 'portal-user';
+        $note['message_status'] = 'New';
+        $note['title'] = 'New Document';
+        foreach ($files["file"] as $file) {
+            $name = $file['name'];
+            $type = $file['type'];
+            $tmp_name = $file['tmp_name'];
+            $size = $file['size'];
+            $data = addNewDocument(
+                $name,
+                $type,
+                $tmp_name,
+                '',
+                $size,
+                $owner,
+                $patient_id,
+                $category_id,
+                '',
+                '',
+                true
+            );
+            if (!empty($data)) {
+                $data['name'] = $name;
+                $note['body'] = xlt('A Portal Patient has uploaded a new document titled') .
+                    ' "' . $name . '" ' .
+                    xlt('to the Documents Onsite Portal Patient category.') . "\n" .
+                    xlt("Please review and take any necessary actions");
+                $messageService->insert($pid, $note);
+            }
+            $rtn[] = $data;
+        }
+        echo json_encode($rtn);
```

Comment:
Since this it not within javascript context, it needs to be escaped (i know it seems like this will break things, but i haven't seen it break anything yet and is done at many different placed in the codebase):
```
echo text(json_encode($rtn));
```

Reply:
this is a return to JS submit but text seems to still work. I don't use it anyway so no foul